src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php

```diff
@@ -1090,6 +1092,41 @@ public function it_updates_a_product_with_a_price_value(): void
         ], $value);
     }
 
+    /** @test */
+    public function it_updates_a_product_with_a_price_value(): void
+    {
+        $this->updateProduct(new SetPriceValue('a_price', null, null, new PriceValue('42', 'EUR')));
+
+        $product = $this->productRepository->findOneByIdentifier('identifier');
+        Assert::assertNotNull($product);
+        $value = $product->getValue('a_price', null, null)->getData()->toArray();
+
+        Assert::assertEqualsCanonicalizing([
+            new ProductPrice('42.00', 'EUR'),
+        ], $value);
+    }
+
+    /** @test */
+    public function it_updates_a_product_with_an_add_price_value(): void
+    {
+        $this->updateProduct(new SetPriceValue('a_price', null, null, new PriceValue('42', 'EUR')));
+        $this->assertProductHasCorrectValueByAttributeCode(
+            'a_price',
+            new PriceCollection([
+                new ProductPrice('42.00', 'EUR')
+            ])
+        );
+
+        $this->updateProduct(new SetPriceValue('a_price', null, null, new PriceValue('50', 'USD')));
+        $this->assertProductHasCorrectValueByAttributeCode(
+            'a_price',
+            new PriceCollection([
+                new ProductPrice('42.00', 'EUR'),
+                new ProductPrice('50.00', 'USD'),
+            ]),
+        );
```

Comment:
In order to test that we update existing price
```suggestion
        $this->updateProduct(new SetPriceValue('a_price', null, null, new PriceValue('42', 'EUR')));
        $this->assertProductHasCorrectValueByAttributeCode(
            'a_price',
            new PriceCollection([
                new ProductPrice('42.00', 'EUR')
            ])
        );

        $this->updateProduct(new SetPriceValue('a_price', null, null, new PriceValue('50', 'USD')));
        $this->assertProductHasCorrectValueByAttributeCode(
            'a_price',
            new PriceCollection([
                new ProductPrice('42.00', 'EUR'),
                new ProductPrice('50.00', 'USD'),
            ]),
        );

        $this->updateProduct(new SetPriceValue('a_price', null, null, new PriceValue('50', 'EUR')));
        $this->assertProductHasCorrectValueByAttributeCode(
            'a_price',
            new PriceCollection([
                new ProductPrice('50.00', 'EUR'),
                new ProductPrice('50.00', 'USD'),
            ]),
        );
```